monkey/monkey_island/cc/ui/src/services/AuthService.js

```diff
@@ -29,16 +33,47 @@ export default class AuthService {
       .then(response => response.json())
       .then(response => {
         if(response.meta.code === 200){
-          this._removeToken();
+          this._removeToken(this.AUTH_TOKEN_NAME_IN_LOCALSTORAGE);
         }
         return response;
       });
   }
 
   authFetch = (url, options) => {
+    this._refreshAuthToken(url, options);
     return this._authFetch(url, options);
   };
 
+  _refreshAuthToken = (url, options) => {
+    this._authFetch(url, options).then(res => {
```

Comment:
After a new token has been acquired, shouldn't this attempt the original request again? Maybe I'm just missing something?

Reply:
See where this is used and what it is returned after it

Reply:
So every call is made twice?

```javascript
  authFetch = (url, options) => {
    this._refreshAuthToken(url, options);
    return this._authFetch(url, options);
  };

  _refreshAuthToken = (url, options) => {
    this._authFetch(url, options).then(res => {
      if(res.status == 401)
      {
        let refreshToken = this._getToken(this.REFRESH_TOKEN_NAME_IN_LOCALSTORAGE);
        if (refreshToken) {
        this._fetchTokenPairFromRefreshToken(this.REFRESH_AUTH_TOKEN_ENDPOINT, {
          method: 'POST',
          body: JSON.stringify({
...
```

Reply:
Why are you calling `_authFetch()` inside `_refreshAuthToken()`? Just check the response status code in `authFetch()`, refresh the token if required, and make the request again.